tests/test_TFUtil.py

```diff
@@ -2783,6 +2783,34 @@ def body(t, _x):
   print("magic (totally arbitrary) res:", session.run(x))
 
 
+def test_flatten_with_seq_len_mask():
+  import numpy as np
+  import numpy.testing as npt
+  batch_dim_axis = 0
+  time_dim_axis = 2
+  x = tf.constant(np.random.rand(10, 16, 11, 16))
+  seq_lens = tf.constant([11, 11, 11, 11, 11, 11, 11, 11, 11, 11])
+
+  x_flat = flatten_with_seq_len_mask(x, seq_lens, batch_dim_axis, time_dim_axis)
+
+  x, x_flat = session.run([x, x_flat])
+  npt.assert_array_almost_equal(x[0, :, 1, :], x_flat[1])
+
+
+def test_flatten_with_unequal_seq_len_mask():
```

Comment:
The function should always start with `test_`, followed by the function name you are testing, i.e. `flatten_with_seq_len_mask`, and then followed by `_whatever_specific_case_you_test`.